# Discover profitable apps in the Apple Store and Google Play Markets

Our company build mobile apps for IOS and Android, all of our apps are free to download, for this reason the number of users determine the revenue of each app.

The goal of this project is analyze data and find powerful insights that can help our developers figure out what type of apps are likely to engage more users.

### Open and explore  Datasets
The first step will be open and explore the both datasets, one containing information about 7,200 IOS apps from de apple store and the other with 10,000 Android apps from Google play.

In [1]:
# Open Apple store Dataset
opened_file = open('AppleStore.csv')
from csv import reader
read_file = reader(opened_file)

ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

# Open Google play Dataset
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)

android = list(read_file)
android_header = android[0]
android = android[1:]

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))


In [3]:
explore_data(ios, 0, 4)
explore_data(android, 0, 4)



['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', 

In [4]:
explore_data(ios, 0,0, True)

Number of rows: 7197
Number of columns: 16


In [5]:
explore_data(android, 0, 0, True)

Number of rows: 10841
Number of columns: 13


In [6]:
print(ios_header)
print('\n')
print(android_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


## Data Cleaning Phase
Delete wrong data

In [7]:
# Check is row 10472 has a missing value

x = len(android[10471])
print(x)

y = len(android[10472])
print(y)

z = len(android[10473])
print(z)


13
12
13


In [8]:
# Compare rows to see which value is being missed

print(android[10471])
print('\n')
print(android[10472]) # does not have value on 'Category' 
print('\n')
print(android[10473])

['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


In [9]:
# Delete the wrong row
print(len(android))
del android[10472]
print(len(android))


10841
10840


## Remove Duplicates
As a part of the cleaning process, we will check ir the Google Play dadaset has duplicate apps

In [10]:
# Example of duplicate data
for app in android:
    name = app[0]
    if name == 'Box':
        print(app)

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


After discover them, it's time to count the number of duplicates:

In [11]:
# Code to count the number of duplicates
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Expected length:', len(android) - 1181)
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:10])

Number of duplicate apps: 1181


Expected length: 9659


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


In [12]:
# Code showing Instagram duplicates with different numbers of reviews  
# at column 4
print(android_header)
print('\n')

for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


As you can see above with the example of "Instagram" app, the duplicates have different number of reviews, this means the higher the reviews, the more recent the duplicate information as been collected.

For this reason we will choose to keep the duplicate with the highest number of reviews, meaning is the most recent one.

### Removing duplicates part 2

We will create a dictionary and use a for loop to save a unique value of each app in the 'reviews_max' dictionary, making sure to store the value withthe highest number of reviews.

With this action we are removing unnecesary duplicates

In [13]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews 
        
print('length dictionary: ', len(reviews_max))

length dictionary:  9659


In [14]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (n_reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)
        
explore_data(android_clean, 0, 3, True)


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


## Remove non-english apps

In [15]:
print(ios[813][1])
print(ios[6731][1])
print('\n')
print(android_clean[4412][0])
print(android_clean[7940][0])


爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


中国語 AQリスニング
لعبة تقدر تربح DZ


As we can see in the example above, there some apps with non-english languages; our focus is in english apps, so will need to remove those apps.

The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the ASCII (American Standard Code for Information Interchange) system.

The first step is to use the 'Ord()' function to find characters not used in english language, this mean outside of the first 127 characters according to the ASCII. 

In [16]:
def english_app(app):
    for a in app:
        if ord(a) > 127:
            return False
        
        
    return True
    
print(english_app('Instagram'))
print(english_app('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english_app('Docs To Go™ Free Office Suite'))
print(english_app('Instachat 😜'))

True
False
False
False


With the current filter, we lose a lot of valuable data because Emojis and speacial characters like '™' have numbers over 127, making them be count as Non-English apps.

To solve these problem, we will improve our filter to just count as Non-English apps, the ones with more than 3 characters with numbers over 127, this will make our filter much more effective.

In [17]:
def english_app(app):
    outside_characters = 0
    for a in app:
        if ord(a) > 127:
            outside_characters += 1
    if outside_characters > 3:
            return False
    else:
        return True

print(english_app('Instagram'))
print(english_app('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english_app('Docs To Go™ Free Office Suite'))
print(english_app('Instachat 😜'))

True
False
True
True


We will use our new function 'english_app' to filter out non-english apps from both datasets.

In [18]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if english_app(name):
        android_english.append(app)

for app in ios:
    name = app[1]
    if english_app(name):
        ios_english.append(app)

        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

## Resume of our Cleaning Process so far

* Removed innacurate data
* Remove duplicate app entries
* Remove non-English apps

Next Step:
* Isolate Free Apps

In [19]:
print(android_header)
print('\n')
explore_data(android_english, 0, 3, True)
print('\n')
print(ios_header)
print('\n')
explore_data(ios_english, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215'

In [20]:
android_final = []
ios_final = []

for app in android_english: 
    price = app[7]
    if price == '0':
        android_final.append(app)
    
    
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
print('Length android final:',len(android_final))
print('Length ios final:',len(ios_final))

Length android final: 8864
Length ios final: 3222


# Start of the Analysis phase

To minimize risks, our validation strategy for an app idea has three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

With that in mind our goal in this analysis is to find apps profiles that are sucessful in both markets (Apple Store and Google Play)

Now that both datasets are clean and ready for analysis is time to start our process to determine the kinds of apps that are likely to attract more users.

# Most Common Apps by Genre

The first step will be define the most common genres in each market. For this, we'll build a frequency table for the 'prime_genre' column of the App Store data set, and the 'Genres' and 'Category' columns of the Google Play data set.

In [21]:
def freq_table(dataset, index):
    table = {}
    total = 0
 # Create Frequency Table 
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
            
  # Create Percentages
    table_percentages = {}
    
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
        
    return table_percentages
 
    # Sort the Percentages

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])  
        


    

### Analysis of Apple Store 

By analyzing the frequency table from the **prime_genre** column; we'll find some insights about the most common genres in the apple store...

In [22]:
display_table(ios_final, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


As we can see above, **Games** is dominating the Apple Store as the most common genre of the free english apps representing **58%** of the total, second comes **Entertainment** with **7.88%** and **Photo & Video** as third with almost **5%**

The vast majority of the free english apps offered in the apple store are focus in offering entertainment, for example 4 of the top 5 most common apps genres are from that area, while more practical apps such as "education", "shopping" or "productivity" are barely seen.

This information can gives a general idea of the most popular apps, but to define a better conclusion is neccesary to analyse not just the offer but the demand of different kind of free english apps in the apple store.



### Analyze of Google Play Market

Now we'll do the same analyses in the freen english Android apps.

In [23]:
display_table(android_final, 1)
print('\n')
display_table(android_final, 9)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

Since the first moment we can see a big difference between IOS and Android markets, for example while the IOS free english apps market is taken over **Games** with almost 60% of the market, in **Android** market we can see a big parity, where the most common category has only **18%** of the market.

If we take a close look at the **Genres** category of the Google Play dataset we can see a big difference compare to the Apple Store, in this case apps for practical purposes have a greater share of the market than apps for entertainment.

For example apps for practical purposes such as "Tools", "Education" or even "Finance" represent 7 of the 8 most common genres in the Google Play Market for free english apps.

The information taken from the analyses of both datasets is valuable but we need to keep exploring to create valuable insights.

# Apple Store... Most popular apps by Genre

In [24]:
genres_ios = freq_table(ios_final, 11)

for genre in genres_ios:
    total = 0
    len_genre = 0
    
    for app in ios_final:
        genre_app = app[11]
        if genre_app == genre:
            number_ratings = float(app[5])
            total += number_ratings
            len_genre += 1

    average_n_ratings = (total / len_genre)
    print(genre, ':', average_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


## Apple Store Insights:

### Let's remember our goal:
* Find the specific type of app that will help our company engage the greatest amount of users.

### Conclusions

* **Games** represent nearly 60% of the market
* The vast majority of Apple Store free english are focus on entertainment
* While apps for practical purpose are rarely seen in the app store even when they get good amount of engagement, So...

This could mean that there's an opportunity for an **app of practical purposes** to be successful in the Apple Store?

Let's dig more into the genres of this area that produce more engagement from users (by number of ratings) to try to find out a specific niche to build our app

### Posibles niches in apps for Practical Purposes 
(by Average Number of Ratings)
* Reference : 74,942
* Weather : 52,279
* Book : 39,758.5
* Finance : 31,467
* Health & Fitness : 23,298 

Of this 5 genres, we can discard "Weather" & "Finance", the first for the difficulty of find reliable data and because people don't spend too much time in those kind of apps, while to build a Finance app we would have to hire a finance expert just for this unique app.

Let's explore the other 3...

In [25]:
print('Reference')
for app in ios_final:
    if app[11] == 'Reference':
        print(app[1], ':', app[5])
        
print('\n')
        

print('Book')
for app in ios_final:
    if app[11] == 'Book':
        print(app[1], ':', app[5])
        
print('\n')
        
print('Health & Fitness')
for app in ios_final:
    if app[11] == 'Health & Fitness':
        print(app[1], ':', app[5])

Reference
Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


Book
Kindle – Read eBooks, Magazines & Textbooks : 252076
Audible – audio books, original series & podcasts : 

These 3 niches show potencial, we could build a good **Reference app** by finding a clear manual/guide in an area that people like to spend time on it, **Health & Fitness** have some potential too but the numbers of engagement is not that high and we would need to hire fitness experts.

# Recommendation... Build a "Book" app

**Why?**
* Has a good amount of engagement --- 39,758.5
* There are only 14 apps representing just 0.43% of the Apple Store 
* The engagement is well distributed between the top 5 apps of the genre (there is not a big monopoly taking over the market)

**& we can...**
## Take the Reading & Learning experience to a whole new level
* Resumes & Powerful insights from the most popular books
* Content with expert advice on "How to Read"
* Chat groups to connect with fellow readers about the book you love

# Google Play... Most Popular apps by Category
The "Genres" column contains too much specific categories making our analysis more difficult, for the moment we'll only focus on the big picture, for instance we'll be working with the **category** column.

In [26]:
categories_android = freq_table(android_final, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace('+', '')
            n_installs = n_installs.replace(',', '')
            n_installs = float(n_installs)
            total += n_installs
            len_category += 1
    average_n_installs = total / len_category
    print(category, ':', average_n_installs)
            
    

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

## Google Play Insights

The 3 most popular categories are:
* Communication --- 38,456,119
* Video Players --- 24,727,872
* Social --- 23,253,652

But this kind of categories are common to be dominated almost completely by a few apps such as "Whatsapp", "Youtube" or "Instagram" (in that order) making difficult to develop a sucessful app in that area.

### Let's remember our goal:
* Recommend a type of app that can be successful in both markets: IOS & Android

We recommended to develop a "Book" app for the Apple Store, let's see if we can do the same for the Google Play Market.

### Analysis of "Books_and_Reference" Category in the Google Play Market
* Have a fair amount of popularity with 8,767,811 average installs
* Represents only 2.14% of the freen english apps in the Google Play Market

This means that a "Books_and_Reference" app have some potential to be sucessful in both markets. Let's dive a littke deeper to comfirm our hypotesis.

We'll look at apps of this category in the middle in terms of popularity, to define more clear the type of app we want to build...



In [28]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

Even when there are quite a few apps focus on Books, we strongly believe that we can take the "Reading & Learning experience" to a whole new level adding a vast amount of innovate functions to make our app unique and be successful in both markets (Apple Store & Google Play Market)

# Final Recommendation

* Develop in both markets an app dedicated to boost the "Reading & Learning Expoerience"

# Conclusions

During this project we analyzed data from the Apple Store (IOS) and from the Google Play Market (Android) with the purpose of recommend a type of app that can be profitable in both markets (IOS and Android)

Our final recommendation is to develop an app focused in take the "Reading & Learning Experience" to a whole new level:

We live in a dynamic world and traditional reading is just one of many ways to enjoy books and learn new things, for example we can add new features like...
 * Short resumes of popular books
 * Newsletters on how to apply books learning to daily life
 * Chat groups where fellow readers can connect
 * Books chapters in video, podcast 